src/request/xhr.ts

```diff
@@ -60,7 +66,7 @@ export default function xhr<T>(url: string, options: XhrRequestOptions = {}): Re
 
 		let timeoutHandle: Handle;
 		request.onreadystatechange = function (): void {
-			if (request.readyState === 4) {
+			if (!isAborted && request.readyState === 4) {
```

Comment:
While the `request` object is unaware of the promise state, it is within closure that has the related promise, named as `promise`, but I agree that the best thing to do is reject and set to a noop.